source/common/http/codec_client.cc

```diff
@@ -158,8 +161,10 @@ CodecClientProd::CodecClientProd(Type type, Network::ClientConnectionPtr&& conne
     break;
   }
   case Type::HTTP3: {
-    // TODO(danzh) Add QUIC codec;
-    NOT_IMPLEMENTED_GCOVR_EXCL_LINE;
+    codec_ = std::unique_ptr<ClientConnection>(dynamic_cast<ClientConnection*>(
```

Comment:
Why do we need a dynamic cast here? Can you have a createQuicHttpClientConnection and createQuicHttpServerConnection on the factory?

Reply:
I'm not concerned about the CPU cost, but the interface/readability cludge that comes from dynamic_cast. IMO it's better to remove it if we can. Yes I would split the factories.